src/esm_parser/esm_parser.py

```diff
@@ -1489,6 +1489,17 @@ def resolve_basic_choose(config, config_to_replace_in, choose_key, blackdict={})
         else:
             choices_available[ckey] = cval
 
+    # Are choices all booleans?
+    choices_bool = True
+    for ckey in choices_available:
+        choices_bool &= isinstance(ckey, bool)
+    # If the choices are booleans and the ``choice`` is a string, try to transform the
+    # try to transform the string in an integer (that will be able to select a choice
+    # from the boolean choices
+    if choices_bool and isinstance(choice, str):
```

Comment:
I am not sure but it might be redundant to check for `if choices_bool`. Simply checking ` if choice == "0" or choice == "1":` might be sufficient since both of these are evaluated as `True`.

Reply:
We want to check if `choices_bool` are boleans because you could have a choose for version numbers:
```
version: 1
choose_version:
     1:
           some_variables: here
     2:
           some_other_vars: here
```
In this case, we should not transform the `version` variable into a boolean because or set of choices are actually version numbers.

Reply:
However, if we have the following:
```
with_wiso: 1
choose_with_wiso:
     true:
           some_variables: here
     false:
           some_other_vars: here
```
Then we want to transform `with_wiso` into an `int` cause and `int` can resolve the boolean choice.